torch/utils/tensorboard/summary.py

```diff
@@ -323,6 +326,85 @@ def scalar(name, tensor, collections=None, new_style=False, double_precision=Fal
         return Summary(value=[Summary.Value(tag=name, simple_value=scalar)])
 
 
+def tensor_proto(tag, tensor):
+    """Outputs a `Summary` protocol buffer containing the full tensor.
+    The generated Summary has a Tensor.proto containing the input Tensor.
+    Args:
+      name: A name for the generated node. Will also serve as the series name in
+        TensorBoard.
+      tensor: Tensor to be converted to protobuf
+    Returns:
+      A tensor protobuf in a `Summary` protobuf.
+    Raises:
+      ValueError: If tensor is too big to be converted to protobuf, or
+                     tensor data type is not supported
+    """
+    if tensor.numel() * tensor.itemsize >= (1 << 31):
+        raise ValueError(
+            "tensor is bigger than protocol buffer's hard limit of 2GB in size"
+        )
+
+    # type maps: torch.Tensor type -> (protobuf type, protobuf val field)
+    type_map = {
+        torch.half: ("DT_HALF", "half_val"),
+        torch.float16: ("DT_HALF", "half_val"),
+        torch.bfloat16: ("DT_BFLOAT", "float_val"),
+        torch.float32: ("DT_FLOAT", "float_val"),
+        torch.float: ("DT_FLOAT", "float_val"),
+        torch.float64: ("DT_DOUBLE", "double_val"),
+        torch.double: ("DT_DOUBLE", "double_val"),
+        torch.int8: ("DT_INT8", "int_val"),
+        torch.uint8: ("DT_UINT8", "int_val"),
+        torch.qint8: ("DT_UINT8", "int_val"),
+        torch.int16: ("DT_INT16", "int_val"),
+        torch.short: ("DT_INT16", "int_val"),
+        torch.int: ("DT_INT32", "int_val"),
+        torch.int32: ("DT_INT32", "int_val"),
+        torch.qint32: ("DT_INT32", "int_val"),
+        torch.int64: ("DT_INT64", "int64_val"),
+        torch.complex32: ("DT_COMPLEX32", "scomplex_val"),
+        torch.chalf: ("DT_COMPLEX32", "scomplex_val"),
+        torch.complex64: ("DT_COMPLEX64", "scomplex_val"),
+        torch.cfloat: ("DT_COMPLEX64", "scomplex_val"),
+        torch.bool: ("DT_BOOL", "bool_val"),
+        torch.complex128: ("DT_COMPLEX128", "dcomplex_val"),
+        torch.cdouble: ("DT_COMPLEX128", "dcomplex_val"),
+        torch.uint8: ("DT_UINT8", "uint32_val"),
+        torch.quint8: ("DT_UINT8", "uint32_val"),
+        torch.quint4x2: ("DT_UINT8", "uint32_val"),
+    }
+
+    if tensor.dtype in type_map:
+        proto_val_field = type_map[tensor.dtype][1]
+        proto_val_contents = (
```

Comment:
For readability, I'd still this make this explicit
proto_val_contents = [] 
if proto_val_field == ...:
    proto_val_contents = ...
elif tensor.numel() == 1
    proto_val_contents  = ...

and so on

Reply:
I see what you mean. Made changes.